module/importer-magento-1/src/Infrastructure/Processor/Step/Magento1GroupedProductProcessor.php

```diff
@@ -8,195 +8,78 @@
-                Assert::notNull($attribute);
-                if (null === $value) {
-                    $record->setValue($field, null);
-                } else {
-                    if (SelectAttribute::TYPE === $type) {
-                        $this->buildSelect($attribute->getId(), $field, $value, $record, $product, $source);
-                    } elseif (MultiSelectAttribute::TYPE === $type) {
-                        $this->buildMultiSelect($attribute->getId(), $field, $value, $record, $product, $source);
-                    } elseif (ImageAttribute::TYPE === $type) {
-                        $this->buildImage($field, $value, $record, $source);
-                    } elseif ($isMultilingual) {
-                        $translation[$source->getDefaultLanguage()->getCode()] = $value;
-                        foreach ($source->getLanguages() as $key => $language) {
-                            if ($product->has($key)) {
-                                $translatedVer = $product->get($key);
-                                if (array_key_exists($field, $translatedVer) && null !== $translatedVer[$field]) {
-                                    $translation[$language->getCode()] = $translatedVer[$field];
-                                }
-                            }
-                        }
-                        $record->setValue($field, new TranslatableStringValue(new TranslatableString($translation)));
-                    } elseif (null !== $value) {
-                        $record->setValue($field, new StringValue($value));
-                    }
-                }
-            }
-
-            $children = $this->getChildren($default);
-
-            if ($children) {
-                $record->set('children', $children);
-            }
-
-            if (null !== $value
-                && '' !== $value
-                && 'children' !== $field
-                && $transformer->hasField($field)
-                && !$record->has($field)
-            ) {
-                $record->set($field, $value);
-            }
-        }
-
-        return $record;
-    }
-
-    /**
-     * @param array $default
-     *
-     * @return null|string
-     */
-    private function getChildren(array $default): ?string
+    private function getChildren(ProductModel $product): array
     {
         $result = [];
 
-        if (array_key_exists('relations', $default)) {
-            $children = explode(',', $default['relations']);
-            $children = array_unique($children);
-
-            foreach ($children as $variant) {
-                $sku = new Sku($variant);
-                $productId = $this->productQuery->findProductIdBySku($sku);
-                if ($productId) {
-                    $result[] = $productId->getValue();
-                }
+        $default = $product->get('default');
+        if (array_key_exists('relations', $default) && null !== $default['relations']) {
+            $variants = explode(',', $default['relations']);
```

Comment:
```
$variants = explode(',', $default['relations']);
foreach ($variants as $variant) {
    $result[] = new Sku($variant);
}
```
That is some kind of pattern.
1. Explode
2. Map to class
Maybe u should create some kind of factory or builder?

```
$result = (new ExplodedCollection($default['relations']))(fn($item) => new Sku($item));
```
?